\hypertarget{genomic-projects-tutorials}{%
\section{Genomic projects tutorials}\label{genomic-projects-tutorials}}

\begin{quote}
:warning: Repository under construction :warning:
\end{quote}

This repository contains a collection of genomic projects that I am
working on. GitHub repository of bioinformatic projects recolving around
genomics using different tools like Plink through \texttt{plinkr} R
package, \texttt{rTASSEL} and TASSEL 5 (GUI), GEMMA for mixed models
analysis in R, SAMtools to analyze BAM files, and other coming soon!

The repository has been created for testing and self-teaching purposes
of biological concept and bioinformatic tools, and make use of other
repositories, scripts and data sources, taken or modified as such.

The report of the studies in progress is at:

\begin{quote}
``Report/build/Genomics\_proj.pdf''
\end{quote}

\hypertarget{contents}{%
\subsection{Contents}\label{contents}}

\begin{itemize}
\tightlist
\item
  \protect\hyperlink{genomic-projects-tutorials}{Genomic projects
  tutorials}

  \begin{itemize}
  \tightlist
  \item
    \protect\hyperlink{contents}{Contents}
  \item
    \protect\hyperlink{tools}{Tools}
  \item
    \protect\hyperlink{example-case-studies}{Example case studies}
  \item
    \protect\hyperlink{resources--data}{Resources \& Data}
  \item
    \protect\hyperlink{setup-of-the-working-environment}{Setup of the
    working environment}

    \begin{itemize}
    \tightlist
    \item
      \protect\hyperlink{get-plink-working-in-linux}{Get \texttt{PLINK}
      working in Linux}

      \begin{itemize}
      \tightlist
      \item
        \protect\hyperlink{get-plinkr-r}{Get \texttt{plinkr}
        (\texttt{R})}
      \end{itemize}
    \item
      \protect\hyperlink{get-tassel-gui-on-linux}{Get \texttt{TASSEL}
      (GUI) on Linux}

      \begin{itemize}
      \tightlist
      \item
        \protect\hyperlink{get-rtassel-r}{Get \texttt{rTASSEL}
        (\texttt{R})}
      \end{itemize}
    \item
      \protect\hyperlink{get-gemma}{Get \texttt{GEMMA}}
    \item
      \protect\hyperlink{get-gapit-r}{Get \texttt{GAPIT} (\texttt{R})}
    \end{itemize}
  \end{itemize}
\end{itemize}

\hypertarget{tools}{%
\subsection{Tools}\label{tools}}

\begin{itemize}
\item
  \textbf{PLINK 1.90} \url{https://www.cog-genomics.org/plink2/}
\item
  \texttt{plinkr} R package repository documentation.
  \url{https://github.com/AJResearchGroup/plinkr}
\item
  \textbf{TASSEL 5} \url{https://www.maizegenetics.net/tassel}.
  \textbf{Bradbury} et al., (2007) TASSEL: software for association
  mapping of complex traits in diverse samples, Bioinformatics, Volume
  23, Issue 19, Pages 2633--2635
  \url{https://doi.org/10.1093/bioinformatics/btm308}
\item
  \texttt{rTASSEL} R package repository documentation. Vignettes:
  \url{https://rtassel.maizegenetics.net/index.html}, Repository:
  \url{https://github.com/maize-genetics/rTASSEL}. \textbf{Monier et
  al.}, (2022). rTASSEL: An R interface to TASSEL for analyzing genomic
  diversity. \emph{Journal of Open Source Software}, 7(76), 4530,
  \url{https://doi.org/10.21105/joss.04530}
\item
  \texttt{GEMMA} Genome-wide Efficient Mixed Model Association
  \url{https://github.com/genetics-statistics/GEMMA}. \textbf{Xiang Zhou
  and Matthew Stephens} (2012). Genome-wide efficient mixed-model
  analysis for association studies. \emph{Nature Genetics} 44, 821--824.
\item
  \texttt{rMVP} A Memory-efficient, Visualization-enhanced, and
  Parallel-accelerated Tool for Genome-Wide Association Study
  https://github.com/xiaolei-lab/rMVP
\item
  \texttt{GPtour} Genomic Prediction in R using Keras models
  https://github.com/miguelperezenciso/GPtour and
  https://keras.posit.co/articles/getting\_started.html
\item
  \texttt{GAPIT} Genome Association and Integrated Tools
  https://github.com/jiabowang/GAPIT
\end{itemize}

\hypertarget{example-case-studies}{%
\subsection{Example case studies}\label{example-case-studies}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\tightlist
\item
  SNP profiling of goat breeds.\emph{Data source}: \textbf{Colli et al.}
  (2018) https://doi.org/10.1186/s12711-018-0422-x
\end{enumerate}

\begin{figure}
\centering
\includegraphics[width=\linewidth]{Figures/goat_mds_12.png}
\caption{Multidimensional scaling of the genotypes}
\end{figure}

\begin{figure}
\centering
\includegraphics[width=\linewidth]{Figures/goat_mds_eigen_filter.png}
\caption{Scree plot of all genotypes and multidimensional scaling of a
subset of genotypes}
\end{figure}

\begin{quote}
\textbf{Multidimensional Scaling (MDS) Plot of a population of 4,653
Individuals from 169 Goat Breeds genotyped with 49,953 SNPs.}

The MDS plot visualizes genetic relationships among 4,653 individuals
from 169 goat breeds. Genetic distances were computed using PLINK to
generate the distance matrix, and MDS analysis was conducted with the
\texttt{cmdscale} function based on genotyping data from 49,953 SNPs.
Each point represents a goat, and spatial arrangement reflects genetic
dissimilarities. This exploratory analysis offers insights into genetic
diversity, population structure, and relatedness.
\end{quote}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  \begin{enumerate}
  \def\labelenumii{\alph{enumii}.}
  \tightlist
  \item
    Manhattan plot of a GWAS on dog population for deafness.\_Data
    source\_: \textbf{Hayward et al.} (2020)
    https://doi.org/10.1371/journal.pone.0232900
  \end{enumerate}
\end{enumerate}

\begin{figure}
\centering
\includegraphics[width=\linewidth]{Figures/manhattan_dogs.png}
\caption{Manhattan plot}
\end{figure}

\begin{figure}
\centering
\includegraphics[width=\linewidth]{Figures/manhattan_filter_acd.png}
\caption{Manhattan plot of a single canine breed}
\end{figure}

\begin{quote}
Manhattan plots showing the genome wide association (GWA) between dog
deafness and their genotype. The plot displays the genomic positions of
single nucleotide polymorphisms (SNPs) across the genome on the x-axis,
with the corresponding -log\textsubscript{10} transformed P-values
indicating the strength of association with the trait on the y-axis. The
red-dashed lines are representation of the 99.99 percentile threshold of
the LOD values.
\end{quote}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  \begin{enumerate}
  \def\labelenumii{\alph{enumii}.}
  \setcounter{enumii}{1}
  \tightlist
  \item
    Plot of the top significant SNPs identified in the above GWAS.
  \end{enumerate}

  Points are jittered around their respective chromosome.
\end{enumerate}

\begin{figure}
\centering
\includegraphics[width=\linewidth]{Figures/top_SNPs_comb.png}
\caption{Top scoring SNPs}
\end{figure}

\begin{quote}
and a zoom in the chromosome 3 above the 99.99 percentile (LOD score =
4.71).
\end{quote}

\begin{figure}
\centering
\includegraphics[width=\linewidth]{Figures/SNP_chr3_top.png}
\caption{Top scoring SNPs of a ABC breed in the 3\textsuperscript{rd}
chromosome}
\end{figure}

\hypertarget{resources-data}{%
\subsection{Resources \& Data}\label{resources-data}}

\begin{itemize}
\item
  \textbf{Marees et al.} (2018) A tutorial on conducting genome-wide
  association studies: Quality control and statistical analysis.
  \emph{Int J Methods Psychiatr Res}. 27:e1608.
  \url{https://doi.org/10.1002/mpr.1608}
\item
  \textbf{Marees et al.} (2018) tutorial
  \url{https://github.com/MareesAT/GWA_tutorial}
\item
  \textbf{Gábor Mészáros} (2021) Genomic Boot Camp Book
  \url{https://genomicsbootcamp.github.io/book/}
\item
  \textbf{Gábor Mészáros} video tutorials
  \url{https://www.youtube.com/c/GenomicsBootCamp}
\item
  \textbf{Colli et al.} (2018) Genome-wide SNP profiling of worldwide
  goat populations reveals strong partitioning of diversity and
  highlights post-domestication migration routes. \emph{Genet Sel Evol}
  50, 58. \url{https://doi.org/10.1186/s12711-018-0422-x}
\item
  DATA: \textbf{Colli et al.} (2020). Signatures of selection and
  environmental adaptation across the goat genome post-domestication
  {[}Dataset{]}. \emph{Dryad}.
  \url{https://doi.org/10.5061/dryad.v8g21pt}
\item
  \textbf{Decker et al.} (2014) Worldwide Patterns of Ancestry,
  Divergence, and Admixture in Domesticated Cattle. \emph{PLOS Genetics}
  10(3):
  e1004254.\href{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004254}{https://doi.org/10.1371/journal.pgen.1004254},
\item
  DATA: \textbf{Decker et al.} (2015) Worldwide patterns of ancestry,
  divergence, and admixture in domesticated cattle {[}Dataset{]}. Dryad.
  \url{https://doi.org/10.5061/dryad.th092}
\end{itemize}

\hypertarget{setup-of-the-working-environment}{%
\subsection{Setup of the working
environment}\label{setup-of-the-working-environment}}

Install R: \href{https://cran.r-project.org/}{The Comprehensive R
Archive Network (CRAN)}

IDE:\href{https://code.visualstudio.com/}{VSCode}\textsuperscript{*}/\href{https://posit.co/download/}{RStudio}\textsuperscript{*}

Install Python:
\href{https://docs.anaconda.com/free/miniconda/index.html}{Miniconda
3}\textsuperscript{*}

OS: Linux\textsuperscript{*}/WSL

\textsuperscript{*}Suggested

\hypertarget{get-plink-working-in-linux}{%
\subsubsection{\texorpdfstring{Get \texttt{PLINK} working in
Linux}{Get PLINK working in Linux}}\label{get-plink-working-in-linux}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Download
  \href{https://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20231211.zip}{PLINK
  1.90 Linux 64-bit}
\item
  Install \texttt{PLINK}
  \texttt{cd\ Downloads/\ \ \ \ \ sudo\ unzip\ plink\_linux\_x86\_64\_20200616.zip\ -d\ plink\_install}
\item
  \texttt{PLINK} in \texttt{usr/local/bin}

\begin{verbatim}
cd plink_install
sudo cp plink /usr/local/bin
sudo chmod 755 /usr/local/bin/plink
\end{verbatim}
\item
  Add \texttt{PLINK} to PATH

  with bash/zsh/\ldots{}

\begin{verbatim}
sudo nano ~/.bashrc
\end{verbatim}

  and include the line:

\begin{verbatim}
export PATH=/usr/local/bin:$PATH
\end{verbatim}

  Save and exit. Refresh the terminal and you should be able to call
  \texttt{plink} from the terminal at any user position in the system.

\begin{verbatim}
source ~/.bashrc
plink --help
\end{verbatim}
\end{enumerate}

\hypertarget{get-plinkr-r}{%
\paragraph{\texorpdfstring{Get \texttt{plinkr}
(\texttt{R})}{Get plinkr (R)}}\label{get-plinkr-r}}

\texttt{PLINK} directly in r.

refer to the installation guide at
https://github.com/AJResearchGroup/plinkr/blob/master/doc/install.md

\begin{verbatim}
library(remotes)
install_github("richelbilderbeek/plinkr")
remotes::install_github("chrchang/plink-ng/2.0/pgenlibr")
library(plinkr)
install_plinks()
\end{verbatim}

\hypertarget{get-tassel-gui-on-linux}{%
\subsubsection{\texorpdfstring{Get \texttt{TASSEL} (GUI) on
Linux}{Get TASSEL (GUI) on Linux}}\label{get-tassel-gui-on-linux}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  Go on the website \url{https://www.maizegenetics.net/tassel} and
  download the last UNIX verison.
\item
  Download the TASSEL\_\{xxx\}\_unix.sh and make it executable

\begin{verbatim}
chmod +x ~/Downloads/TASSEL_{xxx}_unix.sh
\end{verbatim}
\item
  Run the TASSEL installer

\begin{verbatim}
~/Downloads/TASSEL_{xxx}_unix.sh
\end{verbatim}
\end{enumerate}

\hypertarget{get-rtassel-r}{%
\paragraph{\texorpdfstring{Get \texttt{rTASSEL}
(\texttt{R})}{Get rTASSEL (R)}}\label{get-rtassel-r}}

\begin{enumerate}
\def\labelenumi{\arabic{enumi}.}
\item
  \texttt{rJava} installation

\begin{verbatim}
sudo apt install default-jdk
sudo R CMD javareconf
R install.packages("rJava")
\end{verbatim}
\item
  Installation in R

\begin{verbatim}
if (!require("devtools")) install.packages("devtools")
devtools::install_github(
 repo = "maize-genetics/rTASSEL",
 ref = "master",
 build_vignettes = TRUE,
 dependencies = TRUE
)
\end{verbatim}
\item
  Run \texttt{rTASSEL}

  \begin{itemize}
  \tightlist
  \item
    Allocate job's memory\textsuperscript{1} and start the logger (here
    at the root of the project):
  \end{itemize}

  \textsuperscript{1}``-Xmx50g'' and ``-Xms50g'', ``\emph{50g}''
  represents 50 Gigabytes of memory.

  \emph{!! Choose an appropriate value that fits your machine !!}

\begin{verbatim}
options(java.parameters = c("-Xmx50g", "-Xms50g"))
rTASSEL::startLogger(fullPath = NULL, fileName = NULL)
\end{verbatim}

  \begin{itemize}
  \tightlist
  \item
    Run \& infos
  \end{itemize}

\begin{verbatim}
library(rTASSEL)
??rTASSEL
\end{verbatim}

  Useful resource for \texttt{rTASSEL} are the vignettes and tutorials
  at \url{https://rtassel.maizegenetics.net/index.html}
\end{enumerate}

\hypertarget{get-gemma}{%
\subsubsection{\texorpdfstring{Get
\texttt{GEMMA}}{Get GEMMA}}\label{get-gemma}}

\texttt{GEMMA} can be installed from source at the GitHub repo, but is
also available through Bioconda \url{http://www.ddocent.com/bioconda/}.
To install is suggested to have miniconda installed and working, and
then added the channel for Bioconda, you should already have defaults
and conda-forge.

\begin{verbatim}
conda config --add channels defaults
conda config --add channels conda-forge
conda config --add channels bioconda
conda install gemma
\end{verbatim}

And use GEMMA with

\begin{verbatim}
gemma -h
\end{verbatim}

\hypertarget{get-gapit-r}{%
\subsubsection{\texorpdfstring{Get \texttt{GAPIT}
(\texttt{R})}{Get GAPIT (R)}}\label{get-gapit-r}}

R package, here we are going to install it through GitHub. For the
manual visit https://zzlab.net/GAPIT/gapit\_help\_document.pdf

\begin{verbatim}
R> install.packages("devtools")
R> devtools::install_github("jiabowang/GAPIT", force=TRUE)
R> library(GAPIT)
\end{verbatim}

\begin{center}\rule{0.5\linewidth}{0.5pt}\end{center}
