---
title: "rTASSEL"
#author:
output:
  #rmdformats::html_clean:
  rmdformats::downcute:
    downcute_theme: "chaos"
    use_bookdown: false
    self_contained: true
    thumbnails: true
    lightbox: true
    highlight: tango Der Datensatz wurde zwischenzeitlich von einem anderen Benutzer oder in einem anderen Fenster bearbeitet - dieser aktuelle Stand wurde daher neu geladen und ihre ungespeicherten Ã„nderungen evtl. verworfen. 
    gallery: true
    toc_depth: 3
    code_folding: show
    css: css/custom.css
knit:
  (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding =
  encoding, output_dir = "html") })
---

# `rTASSEL`

`rTASSEL` is an extension to TASSEL 5.
```{r}
rm(list = ls())
getwd()
list.files()

library(tidyverse)
knitr::opts_chunk$set(error = TRUE)
```

```{r}
if (!require("rJava")) install.packages("rJava", repos="https://rforge.net")
library(rJava)

if (!require("devtools")) install.packages("devtools")
devtools::install_github("maize-genetics/rTASSEL")
```
Memory allocation for rTASSEL, here we allocate 12 Gb and start the logger
```{r}
#options(java.parameters = c("-Xmx50g", "-Xms50g"))
options(java.parameters = "-Xmx20g")
rTASSEL::startLogger(fullPath = NULL, fileName = NULL)
```
```{r}
library(tidyverse)
library(rTASSEL)
??rTASSEL
help(package = "rTASSEL")
```
### Data import
Test data:
```{r, example}
genoPathVCF <- system.file(
    "extdata",
    "maize_chr9_10thin40000.recode.vcf",
    package = "rTASSEL"
)
genoPathVCF

tasGenoVCF <- rTASSEL::readGenotypeTableFromPath(
  path = genoPathVCF
)
tasGenoVCF
```
Data (goat dryad repo - Colli et al., 2018)

```{r, mydata}
list.files("/home/sscansa/MEGA/R/GWAS/Data/ADAPTmap_genotypeTOP_20160222_full/out/")
getwd()

path <- "/home/sscansa/MEGA/R/GWAS/Data/ADAPTmap_genotypeTOP_20160222_full/out/ADAPTmap_genotypeTOP_20160222_full.vcf"

gen_data <- rTASSEL::readGenotypeTableFromPath(path = path)
gen_data

class(gen_data)
slotNames(gen_data)
```
### Data Filtering and Cleaning

```{r}

```