---
title: "GEMMA"
#author:
output:
  #rmdformats::html_clean:
  rmdformats::downcute:
    downcute_theme: "chaos"
    use_bookdown: false
    self_contained: true
    thumbnails: true
    lightbox: true
    highlight: tango
    gallery: true
    toc_depth: 3
    code_folding: show
    css: css/custom.css
knit:
  (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding =
  encoding, output_dir = "html") })
---

# GWAS analysis with GEMMA

Genome-wide Efficient Mixed Model Association (GEMMA) has the advantage that can use PLINK file directly, without conversions.

```{r}
rm(list = ls())
```

```{r}
library(dplyr)
library(readr)
library(stringr)
library(ggplot2)
library(qqman)
```

# Project 1
## Tutorial on canine deafness

- Code: [https://github.com/GenomicsBootCamp/codeFromGbcVideos/blob/main/GWAS_GEMMA_dogs.r](https://github.com/GenomicsBootCamp/codeFromGbcVideos/blob/main/GWAS_GEMMA_dogs.r)

- *Data*: Hayward, Jessica et al. (2020). A genome-wide association study of deafness in three canine breeds [Dataset]. Dryad. (https://doi.org/10.5061/dryad.sf7m0cg2n).

- *Paper*: Hayward JJ, Kelly-Smith M, Boyko AR, Burmeister L, De Risio L, et al. (2020) A genome-wide association study of deafness in three canine breeds. PLOS ONE 15(5): e0232900. https://doi.org/10.1371/journal.pone.0232900



#### Phenotype data preparation
```{r}
phenoData <- read_table("./Data/deafness_dogs/deafness_pheno.txt")
head(phenoData)
```
Empty variable for the phenotype column in PLINK (later on)
```{r}
PHENOTYPE <- "deafnessCaseControl"
```

making the data usable by plink, the phenotypes ahve ben recoded with 1, 2, and 3.

What happens in the case the phenotype is a continuous variable, for example height, weight or percentage of infection...?

The phenotypee file had to be adapted as plink can only read the phenotypes i n numerical form.
```{r}
system("mkdir ./Data/deafness_dogs/AfterQC")
# recode phenotypes according to methods in the paper
tmp <- phenoData %>%
  mutate(deafnessCode = case_when(
    BAER_test_phenotype == "bilaterally_deaf" ~ "1",
    BAER_test_phenotype == "unilaterally_deaf" ~ "2",
    BAER_test_phenotype == "hearing" ~ "3"
  )) %>%
  mutate(deafnessCaseControl = case_when(
    BAER_test_phenotype == "bilaterally_deaf" ~ "1",
    BAER_test_phenotype == "hearing" ~ "2"
  )) %>%
  mutate(FID = dogID) %>%
  rename(IID = dogID) %>%
#  drop_na() %>%
  select(FID, IID, deafnessCode, deafnessCaseControl) %>%
  write_delim("./Data/deafness_dogs/AfterQC/GWAS_pheno_renamed.txt", delim = " ")
```
### Plink QC Genotype

here we use the flag `--chr-set 38` instad of the equivalent flag `--dog`.

`--nonfounders` does not take in consideration the pedigree (parental) information in the .fam file, because there is no parental information and therefore information has to be removed otherwine all the individuals (animals/plants/...) are treated as founders, i.e., individuals who establish a new population or breeding group.

```{r}
data_in <- "./Data/deafness_dogs/deafness"
data_QC <- "./Data/deafness_dogs/AfterQC/deafness_QC"

system(str_c("
plink \\
--chr-set 38 \\
--nonfounders \\
--bfile ", data_in, " \\
--autosome \\
--mind 0.1 \\
--geno 0.05 \\
--make-bed \\
--out ", data_QC, sep = ""))
```

the data folder AfterQC contains all the data after quality filtering.

### Merge Genotype - Phenotype table in PLINK

```{r}
GWAS_pheno <- "./Data/deafness_dogs/AfterQC/GWAS_pheno_renamed.txt"
GEMMA_input <- "./Data/deafness_dogs/AfterQC/GEMMA_input"

system(str_c("
plink \\
--chr-set 38 \\
--bfile ", data_QC, " \\
--nonfounders \\
--allow-no-sex \\
--pheno ", GWAS_pheno, " \\
--pheno-name ", PHENOTYPE, " \\
--make-bed \\
--out ", GEMMA_input, sep = ""))
```

Now that we have created the phenotypes in the .fam file we can proceed to run the association genotype-phenotype with GEMMA.

#### Optional filtering

```{r}
# filter of a single dog breed

# select only Australian cattle dogs - direct comparison w the paper
#phenoData %>%
#  filter(breed == "australian_cattle_dog") %>%
#  mutate(FID = dogID) %>%
#  rename(IID = dogID) %>%
#  select(FID, IID) %>%
#  write_delim("dogsToKeep.txt", delim = " ")

# Update the .fam file with a single dog breed as GEMMA input
#system(str_c("./plink --dog --bfile inputForGemma --nonfounders --allow-no-sex ",
#             " --keep dogsToKeep.txt --make-bed --out inputForGemma_ACD"))
```

### Run GWAS with GEMMA

#####Filtered Dataset

```{r}
# select only Australian cattle dogs - direct comparison w the paper
#phenoData %>%
#  filter(breed == "australian_cattle_dog") %>%
#  mutate(FID = dogID) %>%
#  rename(IID = dogID) %>%
#  select(FID, IID) %>%
#  write_delim("dogsToKeep.txt", delim = " ")

#system(str_c("./plink --dog --bfile inputForGemma --nonfounders --allow-no-sex ",
#             " --keep dogsToKeep.txt --make-bed --out inputForGemma_ACD"))
```

##### Complete Dataset

on the output flag `-o` seems there is no way to redirect the output files in a specific folder, either it will create a outut folder at the project root or it will pop an error messsage, strange.
However, the folder that has been created is then moved to the "/Data/deafness_dogs/AfterQC/output/" where it belongs.

```{r}
# compute the relatedness matrix for population structure correction

#RelMat <- "Data/deafness_dogs/AfterQC/RelMat/"

system(str_c("
gemma \\
-bfile ", GEMMA_input, " \\
-gk 1 \\
-o RelMat"))

system("mv output/ Data/deafness_dogs/AfterQC")

# run GEMMA
system(str_c("
gemma \\
-bfile Data/deafness_dogs/AfterQC/GEMMA_input \\
-k Data/deafness_dogs/AfterQC/output//RelMat.cXX.txt \\
-lmm 2 \\
-o GWASresults.lmm"))

system("rsync -av output/ Data/deafness_dogs/AfterQC/output/")
system("rm -r output/")

# in some cases the output file could contain incorrect line breaks
# possible confusion of Windows and Linux line endings
# in such cases run the following line - and than change the file name loaded for data vizualization
#system("tr -d '\r' <GWASresults.lmm.assoc.txt > GWASresults.lmm.assoc.lineBreaksOk.txt")
```

From GEMMA's manual: "You can use `-outdir` with GEMMA as a bash script

`-outdir $(mktemp -d -p $HOME)`

makes a unique temp directory<sup>*</sup> where the output is stored, here relative to $HOME, but you can take any path.""

<sup>*</sup> Errata Corriege: The `mktemp` command creates a temporary filename (not a directory) which does not exist yet. The `-p` option tells to redirect in a directory.

* Now we have as output the results of the GWAS analysis and we can plot it. The  next code will load the necessary packages and then present


```{r}
# read in the GWAS results
resultGemma <- read_table("Data/deafness_dogs/AfterQC/output/GWASresults.lmm.assoc.txt")
head(resultGemma)
```
```{r}
# cumulative position of SNPs
data_man <- resultGemma %>%
  # chromosome sizes
  group_by(chr) %>%
  summarise(chr_len = max(ps)) %>%
  # cumulative position of chromosomes
  mutate(tot = cumsum(chr_len) - chr_len) %>%
  select(-chr_len) %>%
  # add to the gemma dataset
  left_join(., resultGemma, by = c("chr" = "chr")) %>%
  arrange(chr, ps) %>%
  mutate(PScum = ps + tot,
  neglogP = -log10(p_lrt))  %>%
  select(chr, tot, ps, PScum, p_lrt, neglogP)
```
```{r}
# positions where the thick marks of the chromosomes names will go
x_axis_chr <- data_man %>%
  group_by(chr) %>%
  summarise(center = (max(PScum) + min(PScum)) / 2)
```
```{r}
# compute the Bonferroni threshold
bonferroni05 <- -log10(0.05 / nrow(resultGemma))
bonferroni01 <- -log10(0.01 / nrow(resultGemma))
tresh <- -log10(5 * 10^(-8))

plot_mann <- data_man %>%
  ggplot(aes(PScum, neglogP)) +
  geom_point(aes(color = as.factor(chr)), size = 2, alpha = .4, show.legend = FALSE) +
  geom_hline(yintercept = bonferroni05, linetype = "dashed") +
  #geom_hline(yintercept = bonferroni01, linetype = "dashed") +
  scale_color_manual(values = rep(c("firebrick", "steelblue"), 38)) +
  scale_x_continuous(label = x_axis_chr$chr, breaks = x_axis_chr$center) +
  scale_y_continuous(expand = c(0, 0)) +
  labs(
    title = "manhattan_plot",
    x = "Chromosome #",
    y = bquote(-log[10](italic(P)))) +
  theme_bw() +
    theme(
      panel.border = element_blank(),
      panel.grid.minor.x = element_blank(),
      panel.grid.major.x = element_blank(),
      aspect.ratio = 0.6
    )

path_manhattan <- "Figures/manhattan_dogs.png"

ggsave(path_manhattan , plot = plot_mann, width = 10)
knitr::plot_crop(path_manhattan)

```

this is a great resource for manhattan plot customisation [https://r-graph-gallery.com/101_Manhattan_plot.html](https://r-graph-gallery.com/101_Manhattan_plot.html)

```{r}
library(qqman)

png("Figures/qqman.png")
  qq(data_man$p_lrt)
dev.off()

```
is it possible to merge also the names of the canine breeds and run it through a geom_wrap?

##### Select the most significant SNPs'

```{r}
resultGemma %>%
  mutate(negLogP = -log10(p_lrt)) %>%
  select(chr, rs, p_lrt, negLogP) %>%
  summary()
  filter(negLogP > 5)
```