---
title: "Vitis"
#author:
output:
  #rmdformats::html_clean:
  rmdformats::downcute:
    downcute_theme: "chaos"
    use_bookdown: false
    self_contained: true
    thumbnails: true
    lightbox: true
    highlight: tango
    gallery: true
    toc_depth: 3
    code_folding: show
    css: css/custom.css
knit:
  (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding =
  encoding, output_dir = "html") })
---

# Genomic project to investigate  the genetic basis of complex traits in Vitis vinifera.

## Introduction

The aim of this project is to identify and characterize genes associated with important plant traits using high-throughput sequencing technologies.

## Data
Here we analyse the data of the paper:
Smith HM, Smith BP, Morales NB, Moskwa S, Clingeleffer PR, Thomas MR (2018) SNP markers tightly linked to root knot nematode resistance in grapevine (Vitis cinerea) identified by a genotyping-by-sequencing approach followed by Sequenom MassARRAY validation. PLoS ONE 13 (2): e0193121. https://doi.org/10.1371/journal.pone.0193121

Data can be found in the DRYAD repository:
Smith, Harley M. et al. (2019). Data from: SNP markers tightly linked to root knot nematode resistance in grapevine (Vitis cinerea) identified by a genotyping-by-sequencing approach followed by Sequenom MassARRAY validation [Dataset]. Dryad. https://doi.org/10.5061/dryad.1d7n9

After downloading and unzipping the files, I used `rTASSEL` to run a quality control on the file `all.taxamerged.filtered.recode.vcf` which is described in the *README.txt* to be the data from the sequencing. This file contains SNP data for C2-50, Riesling and the 90 C2-50 x Riesling F1 individuals.

In `rTASSEL`, i import the `all.taxamerged.filtered.recode.vcf` file

```{r}
rm(list = ls())
```
```{r}
library(rTASSEL)
library(tidyverse)
# if (!require("BiocManager", quietly = TRUE))
#     install.packages("BiocManager")
# BiocManager::install("ggtree")
library(ggtree)
```
```{r}
pathVCF <- "Data/SNP_linked_rootknot_nematode_resistance_grapevine/SNP_data/all.taxamerged.filtered.recode.vcf"
tassel_in_geno_vcf_all <- readGenotypeTableFromPath(pathVCF)
 ```

 ```{r}
pathVCF <- "Data/SNP_linked_rootknot_nematode_resistance_grapevine/SNP_data/Riesling_2973_SNPset.vcf"
tassel_in_geno_vcf_riesling <- readGenotypeTableFromPath(pathVCF)
 ```

  ```{r}
pathVCF <- "Data/SNP_linked_rootknot_nematode_resistance_grapevine/SNP_data/C2_50_3974_SNPset.vcf"
tassel_in_geno_vcf_c2 <- readGenotypeTableFromPath(pathVCF)
 ```
 ```{r}
#pathPheno <- "Data/SNP_linked_rootknot_nematode_resistance_grapevine/SNP_data/C2_50_367_MJR1.out.txt"
#read_table(pathPheno)
#tassel_in_pheno <- readPhenotypeFromPath(pathPheno)
 ```
 ```{r}
 genoMat <- as.matrix(tassel_in_geno_vcf_all)
 genoMat
 ```