---
title: "Wheat"
#author:
output:
  #rmdformats::html_clean:
  rmdformats::downcute:
    downcute_theme: "chaos"
    use_bookdown: false
    self_contained: true
    thumbnails: true
    lightbox: true
    highlight: tango
    gallery: true
    toc_depth: 3
    code_folding: show
    css: css/custom.css
knit:
  (function(inputFile, encoding) { rmarkdown::render(inputFile, encoding =
  encoding, output_dir = "html") })
---

# Genomic project to investigate  the genetic basis of complex traits in Wheat (*T. aestivum*).

```{r}
rm(list = ls())
```

## Introduction

## Durum wheat (*Triticum turgidum* subsp. *durum*)

Data:
Dell'Acqua, Matteo (2022). Characterization data for the EtNAM population and Ethiopian durum wheat landraces diversity panel [Dataset]. Dryad. https://doi.org/10.5061/dryad.w6m905qrv form the orginal research article https://onlinelibrary.wiley.com/doi/10.1111/pbi.12538


Here we deal with data stored in the .Rdagst data format, great!

In the README_file.txt of the data repository we ahve the folllowing description:

>The file "*diversity.panel.data.gp.Rdata*" contains molecular and phenotypic data
relative to the Ethiopian durum wheat diversity panel used in this study.
The Rdata file contains serveral R objects:
>
>- **info**, passport data for the accessions included in the study
>- **geno**, containing genotypic data scored on 90K markers on the wheat array
>- **snp.pos**, containing estimated genomic positions of the 90K markers
>- **met**, containing agronomic characterization data
>- **farm**, containing participatory variety selection data
>
>The file "*etnam.data.gp.Rdata*" contains molecular and phenotypic data
relative to the EtNAM population used in this study.
The Rdata file contains serveral R objects:
>
>- **infonam**, passport data for the accessions included in the study
>- **infofound**, passport data for the parental lines of the EtNAM
>- **genonam**, containing genotypic data scored on 13K markers on the wheat array
>- **snp.pos.nam**, containing estimated genomic positions of the 90K markers
>- **metnam**, containing agronomic characterization data
>- **farmnam**, containing participatory variety selection data

```{r, echo = FALSE}
library(tidyverse)
library(corrplot)
```
```{r}
data_et <- load("Data/Triticum-durum/etnam.data.gp.Rdata")
data_et
```

```{r}
data <- load("Data/Triticum-durum/diversity.panel.data.gp.Rdata")
data
# agronomic characterization data
met_data <- met
# participatory variety selection data
farm_data <- farm
# genotypic data scored on 90K markers on the wheat array
geno_data <- geno
# estimated genomic positions of the 90K markers
snp_pos_data <- snp.pos
# passport data for the accessions included in the study
info_data <- info
```

## Exploratory data analysis
### Agronomic data
```{r}
met_data %>% head()
```
ID = identification
DB = days to booting (d)
DF = days to flowering (d)
DM = days to maturity (d)
PH = plant height (cm)
NET = number of effective tillers per plant (n)
SPL = spike length (cm)
SPS = number of seeds per spike (n)
BM = biomass (t/ha)
GY = grain yeld (t/ha)
TGW = thousand grain weight (g)

##### Correlation Matrix:
```{r}
unique_ids <- make.unique(met_data$ID)

rownames(met_data) <- unique_ids

eda_met <- met_data %>%
  select(DB:ncol(met_data))

summary(eda_met)
nrow(eda_met)

eda_met <- na.omit(eda_met)

eda_met_corr <- cor(eda_met)

corrplot(eda_met_corr, method = "color", type = "lower", order = "hclust", tl.col = "black", tl.srt = 90)

```

```{r}
# function cor.mtest to pass on a matrix  and perform the correlation test using the Pearson correlation coefficient as metric.
# if Spearman or Kendall correlations are needed, then change the method argument in cor.test.
cor.mtest <- function(mat, ...) {
    mat <- as.matrix(mat)
    n <- ncol(mat)
    p.mat<- matrix(NA, n, n)
    diag(p.mat) <- 0
    for (i in 1:(n - 1)) {
        for (j in (i + 1):n) {
            tmp <- cor.test(mat[, i], mat[, j], ...)
            p.mat[i, j] <- p.mat[j, i] <- tmp$p.value
        }
    }
  colnames(p.mat) <- rownames(p.mat) <- colnames(mat)
  p.mat
}
```
```{r}
eda_met <- as.matrix(eda_met)

pearson_p <- cor.mtest(eda_met)
head(pearson_p[, 1:5])

# col <- colorRampPalette(c("#BB4444", "#EE9988", "#FFFFFF", "#77AADD", "#4477AA"))

h <- 700
png("Figures/wheat_corrplot.png", width = h, height = h, res = 120)

corrplot(
  eda_met_corr,
  method = "color",
  #col=col(200),
  type = "upper",
  order = "hclust",
  tl.col = "black",
  tl.srt = 90,
  diag = F,
  p.mat = pearson_p,
  sig.level = 0.01,
  addCoef.col = "grey10"
)

dev.off()

```


##### PCA
To reveal hidden structures in the population.

| prcomp() | name     | Description                                                                  |
|----------|----------|------------------------------------------------------------------------------|
| sdev 	   | sdev 	  | standard deviations of the principal components                              |
| rotation | loadings | matrix of variable loadings (columns are eigenvectors)                       |
| center 	 | center 	| variable means (means that were substracted)                                 |
| scale 	 | scale 	  | variable standard deviations (the scaling applied to each variable)          |
| x 	     | scores 	| coordinates of the individuals (observations) on the principal components.   |

```{r}
# using the eda_met data, after removing the NAs
pca <- prcomp(eda_met, center = T, scale = T)
pca
```

Perform SVD:
```{r}
# standardize data
std_eda_met <- scale(eda_met)
# cov matrix
cov_mat <- cov(std_eda_met)
# singular value decomposition
svd_res <- svd(cov_mat)
# we obrain the matrix $d, $u and $v: U and V are the orthogonal
```

We obtain the decomposition of the covariate matrix as:

$ X=UDV^{T} $

Where $U$ and $V$ are the orthogonal and $V^{T}$ the transposed.
And $D$ is the diagonal matrix with the singular values.
also represented as $D=U^{T}XV$

with the function svd() we get the matrices $d which has the singular $X$ values [min(n.p)], $u with the left singular vectors [c(n,nu)], v with the right singlular vector of $X$ [c(p,nv)].

Extract the values:
```{r}
# Extract singular values, left singular vectors (loadings), and right singular vectors (scores)
singular_values <- svd_res$d
loadings <- svd_res$u
scores <- std_eda_met %*% svd_res$v
```
```{r}
# Principal component loadings
loadings
# Principal component scores
scores
# Variance explained by each principal component
var_explained
```

And perform PCA
```{r}
n_PCs <- 1:length(var_explained)

exp_var_df <- tibble::tibble(PC = n_PCs) %>%
  mutate(exp_var = var_explained, exp_var_perc = round(exp_var * 100, 2))

plot_exp_var <-
  exp_var_df %>%
  ggplot(., aes(factor(PC), exp_var_perc)) +
  geom_col(alpha = .6, fill = "steelblue") +
  geom_point(size = 2) +
  geom_line(aes(x = PC)) +
  geom_text(aes(PC + 0.4, exp_var_perc + 4, label = paste0(exp_var_perc, "%")), angle = 45) +
  scale_y_continuous(expand = c(0, 0), limits = c(0, 72)) +
  labs(x = "Dimension", y = "Explained Variances") +
  theme_classic() +
  theme(aspect.ratio = .75)


path <- "Figures/wheat_exp_var_agro_data.png"

ggsave(path, plot = plot_exp_var, width = 10, dpi = 120)
knitr::plot_crop(path)
```

