\markdownRendererDocumentBegin
\markdownRendererSectionBegin
\markdownRendererHeadingOne{Genomic projects tutorials}\markdownRendererInterblockSeparator
{}!!! warning "This repository is under construction"\markdownRendererInterblockSeparator
{}This repository contains a collection of genomic projects that I am working on. GitHub repository of bioinformatic projects recolving around genomics using different tools like Plink through \markdownRendererCodeSpan{plinkr} R package, \markdownRendererCodeSpan{rTASSEL} and TASSEL 5 (GUI), GEMMA for mixed models analysis in R, SAMtools from the command line to analyze BAM files, gBLUP coming soon.\markdownRendererInterblockSeparator
{}The repository has been created for testing and self-teaching purposes of biological concept and bioinformatic tools, and make use of other repositories, scripts and data sources, taken or modified as such.\markdownRendererInterblockSeparator
{}\markdownRendererOlBeginTight
\markdownRendererOlItemWithNumber{1}SNP profiling of goat breeds.<br>\markdownRendererEmphasis{Data source}: \markdownRendererStrongEmphasis{Colli et al.} (2018) https://doi.org/10.1186/s12711-018-0422-x\markdownRendererOlItemEnd 
\markdownRendererOlEndTight \markdownRendererInterblockSeparator
{}<img src="Figures/goat\markdownRendererEmphasis{mds}2.png" height="350px">\markdownRendererInterblockSeparator
{}<small>\markdownRendererStrongEmphasis{Multidimensional Scaling (MDS) Plot of a population of 4,653 Individuals from 169 Goat Breeds genotyped with 49,953 SNPs.}</small>\markdownRendererInterblockSeparator
{}<small>The MDS plot visualizes genetic relationships among 4,653 individuals from 169 goat breeds. Genetic distances were computed using PLINK to generate the kinship matrix, and MDS analysis was conducted with the \markdownRendererCodeSpan{cmdscale} function based on genotyping data from 49,953 SNPs. Each point represents a goat, and spatial arrangement reflects genetic dissimilarities. This exploratory analysis offers insights into genetic diversity, population structure, and relatedness. </small>\markdownRendererInterblockSeparator
{}\markdownRendererOlBeginTight
\markdownRendererOlItemWithNumber{1}a. Manhattan plot of a GWAS on dog population for deafness.\markdownRendererEmphasis{Data source}: \markdownRendererStrongEmphasis{Hayward et al.} (2020) https://doi.org/10.1371/journal.pone.0232900\markdownRendererOlItemEnd 
\markdownRendererOlEndTight \markdownRendererInterblockSeparator
{}<img src="Figures/manhattan\markdownRendererEmphasis{dogs.png" height="350px"> <img src="Figures/manhattan}filter\markdownRendererUnderscore{}acd.png" height="350px"> <small>Manhattan plots showing the genome wide association (GWA) between dog deafness and their genotype. The plot displays the genomic positions of single nucleotide polymorphisms (SNPs) across the genome on the x-axis, with the corresponding -log<sub>10</sub> transformed P-values indicating the strength of association with the trait on the y-axis. The red-dashed lines are representation of the 99.99 percentile treshold of the LOD values.</small>\markdownRendererInterblockSeparator
{}\markdownRendererOlBeginTight
\markdownRendererOlItemWithNumber{2}b. Plot of the top significant SNPs identified in the above GWAS.\markdownRendererOlItemEnd 
\markdownRendererOlEndTight \markdownRendererInterblockSeparator
{}<img src="Figures/top\markdownRendererEmphasis{SNPs}comb.png" height="350px">\markdownRendererInterblockSeparator
{}and a zoom in the chromosome 3 above LOD 3.5.\markdownRendererInterblockSeparator
{}<img src="Figures/SNP\markdownRendererEmphasis{chr3}top.png" height="350px">\markdownRendererInterblockSeparator
{}\markdownRendererSectionBegin
\markdownRendererHeadingTwo{Tools}\markdownRendererInterblockSeparator
{}\markdownRendererUlBegin
\markdownRendererUlItem \markdownRendererStrongEmphasis{PLINK 1.90} \markdownRendererLink{https://www.cog-genomics.org/plink2/}{https://www.cog-genomics.org/plink2/}{https://www.cog-genomics.org/plink2/}{}\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererCodeSpan{plinkr} R package repository documentation. \markdownRendererLink{https://github.com/AJResearchGroup/plinkr}{https://github.com/AJResearchGroup/plinkr}{https://github.com/AJResearchGroup/plinkr}{}\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererStrongEmphasis{TASSEL 5} \markdownRendererLink{https://www.maizegenetics.net/tassel}{https://www.maizegenetics.net/tassel}{https://www.maizegenetics.net/tassel}{}. \markdownRendererStrongEmphasis{Bradbury} et al., (2007) TASSEL: software for association mapping of complex traits in diverse samples, Bioinformatics, Volume 23, Issue 19, Pages 2633–2635 \markdownRendererLink{https://doi.org/10.1093/bioinformatics/btm308}{https://doi.org/10.1093/bioinformatics/btm308}{https://doi.org/10.1093/bioinformatics/btm308}{}\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererCodeSpan{rTASSEL} R package repository documentation. <br> Vignettes: \markdownRendererLink{https://rtassel.maizegenetics.net/index.html}{https://rtassel.maizegenetics.net/index.html}{https://rtassel.maizegenetics.net/index.html}{}, Repository: \markdownRendererLink{https://github.com/maize-genetics/rTASSEL}{https://github.com/maize-genetics/rTASSEL}{https://github.com/maize-genetics/rTASSEL}{}. \markdownRendererStrongEmphasis{Monier et al.}, (2022). rTASSEL: An R interface to TASSEL for analyzing genomic diversity. \markdownRendererEmphasis{Journal of Open Source Software}, 7(76), 4530, \markdownRendererLink{https://doi.org/10.21105/joss.04530}{https://doi.org/10.21105/joss.04530}{https://doi.org/10.21105/joss.04530}{}\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererCodeSpan{GEMMA} Genome-wide Efficient Mixed Model Association \markdownRendererLink{https://github.com/genetics-statistics/GEMMA}{https://github.com/genetics-statistics/GEMMA}{https://github.com/genetics-statistics/GEMMA}{}. \markdownRendererStrongEmphasis{Xiang Zhou and Matthew Stephens} (2012). Genome-wide efficient mixed-model analysis for association studies. \markdownRendererEmphasis{Nature Genetics} 44, 821–824.\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererCodeSpan{rMVP} A Memory-efficient, Visualization-enhanced, and Parallel-accelerated Tool for Genome-Wide Association Study https://github.com/xiaolei-lab/rMVP\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererCodeSpan{GPtour} Genomic Prediction in R using Keras models https://github.com/miguelperezenciso/GPtour and https://keras.posit.co/articles/getting\markdownRendererUnderscore{}started.html\markdownRendererUlItemEnd 
\markdownRendererUlEnd \markdownRendererInterblockSeparator
{}
\markdownRendererSectionEnd \markdownRendererSectionBegin
\markdownRendererHeadingTwo{Resources \markdownRendererAmpersand{} Data}\markdownRendererInterblockSeparator
{}\markdownRendererUlBegin
\markdownRendererUlItem \markdownRendererStrongEmphasis{Marees et al.} (2018) A tutorial on conducting genome-wide association studies: Quality control and statistical analysis. \markdownRendererEmphasis{Int J Methods Psychiatr Res}. 27:e1608. \markdownRendererLink{https://doi.org/10.1002/mpr.1608}{https://doi.org/10.1002/mpr.1608}{https://doi.org/10.1002/mpr.1608}{}\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererStrongEmphasis{Marees et al.} (2018) tutorial \markdownRendererLink{https://github.com/MareesAT/GWA\markdownRendererUnderscore{}tutorial}{https://github.com/MareesAT/GWA\markdownRendererUnderscore{}tutorial}{https://github.com/MareesAT/GWA_tutorial}{}\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererStrongEmphasis{Gábor Mészáros} (2021) Genomic Boot Camp Book \markdownRendererLink{https://genomicsbootcamp.github.io/book/}{https://genomicsbootcamp.github.io/book/}{https://genomicsbootcamp.github.io/book/}{}\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererStrongEmphasis{Gábor Mészáros} video tutorials \markdownRendererLink{https://www.youtube.com/c/GenomicsBootCamp}{https://www.youtube.com/c/GenomicsBootCamp}{https://www.youtube.com/c/GenomicsBootCamp}{}\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererStrongEmphasis{Colli et al.} (2018) Genome-wide SNP profiling of worldwide goat populations reveals strong partitioning of diversity and highlights post-domestication migration routes. \markdownRendererEmphasis{Genet Sel Evol} 50, 58. \markdownRendererLink{https://doi.org/10.1186/s12711-018-0422-x}{https://doi.org/10.1186/s12711-018-0422-x}{https://doi.org/10.1186/s12711-018-0422-x}{}\markdownRendererUlItemEnd 
\markdownRendererUlItem DATA: \markdownRendererStrongEmphasis{Colli et al.} (2020). Signatures of selection and environmental adaptation across the goat genome post-domestication [Dataset]. \markdownRendererEmphasis{Dryad}. \markdownRendererLink{https://doi.org/10.5061/dryad.v8g21pt}{https://doi.org/10.5061/dryad.v8g21pt}{https://doi.org/10.5061/dryad.v8g21pt}{}\markdownRendererUlItemEnd 
\markdownRendererUlItem \markdownRendererStrongEmphasis{Decker et al.} (2014) Worldwide Patterns of Ancestry, Divergence, and Admixture in Domesticated Cattle. \markdownRendererEmphasis{PLOS Genetics} 10(3): e1004254.\markdownRendererLink{https://doi.org/10.1371/journal.pgen.1004254}{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004254}{https://journals.plos.org/plosgenetics/article?id=10.1371/journal.pgen.1004254}{},\markdownRendererUlItemEnd 
\markdownRendererUlItem DATA: \markdownRendererStrongEmphasis{Decker et al.} (2015) Worldwide patterns of ancestry, divergence, and admixture in domesticated cattle [Dataset]. Dryad. \markdownRendererLink{https://doi.org/10.5061/dryad.th092}{https://doi.org/10.5061/dryad.th092}{https://doi.org/10.5061/dryad.th092}{}\markdownRendererUlItemEnd 
\markdownRendererUlEnd \markdownRendererInterblockSeparator
{}
\markdownRendererSectionEnd \markdownRendererSectionBegin
\markdownRendererHeadingTwo{Setup of the working environment}\markdownRendererInterblockSeparator
{}Install R: \markdownRendererLink{The Comprehensive R Archive Network (CRAN)}{https://cran.r-project.org/}{https://cran.r-project.org/}{}\markdownRendererInterblockSeparator
{}IDE:\markdownRendererLink{VSCode}{https://code.visualstudio.com/}{https://code.visualstudio.com/}{}/\markdownRendererLink{RStudio}{https://posit.co/download/}{https://posit.co/download/}{}<sup>*</sup>\markdownRendererInterblockSeparator
{}Install Python: \markdownRendererLink{Miniconda 3}{https://docs.anaconda.com/free/miniconda/index.html}{https://docs.anaconda.com/free/miniconda/index.html}{}<sup>*</sup>\markdownRendererInterblockSeparator
{}OS: Linux<sup>*</sup>/WSL\markdownRendererInterblockSeparator
{}<small><sup>*</sup>Suggested</small>\markdownRendererInterblockSeparator
{}\markdownRendererSectionBegin
\markdownRendererHeadingThree{Get PLINK working in Linux}\markdownRendererInterblockSeparator
{}\markdownRendererOlBegin
\markdownRendererOlItemWithNumber{1}Download \markdownRendererLink{PLINK 1.90 Linux 64-bit}{https://s3.amazonaws.com/plink1-assets/plink\markdownRendererUnderscore{}linux\markdownRendererUnderscore{}x86\markdownRendererUnderscore{}64\markdownRendererUnderscore{}20231211.zip}{https://s3.amazonaws.com/plink1-assets/plink_linux_x86_64_20231211.zip}{}\markdownRendererOlItemEnd 
\markdownRendererOlItemWithNumber{2}Install PLINK \markdownRendererCodeSpan{
cd Downloads/
sudo unzip plink\markdownRendererUnderscore{}linux\markdownRendererUnderscore{}x86\markdownRendererUnderscore{}64\markdownRendererUnderscore{}20200616.zip -d plink\markdownRendererUnderscore{}install
}\markdownRendererOlItemEnd 
\markdownRendererOlItemWithNumber{3}PLINK in \markdownRendererCodeSpan{usr/local/bin}\markdownRendererInterblockSeparator
{}\markdownRendererCodeSpan{
cd plink\markdownRendererUnderscore{}install
sudo cp plink /usr/local/bin
sudo chmod 755 /usr/local/bin/plink
}\markdownRendererOlItemEnd 
\markdownRendererOlItemWithNumber{4}Add PLINK to PATH\markdownRendererInterblockSeparator
{}with bash/zsh/...\markdownRendererInterblockSeparator
{}\markdownRendererCodeSpan{
sudo nano \markdownRendererTilde{}/.bashrc
}\markdownRendererInterblockSeparator
{}adn include the line:\markdownRendererInterblockSeparator
{}\markdownRendererCodeSpan{
export PATH=/usr/local/bin:\markdownRendererDollarSign{}PATH
}\markdownRendererInterblockSeparator
{}Save and exit. Refresh the terminal and you should be able to call \markdownRendererCodeSpan{plink} from the terminal at any user position in the system.\markdownRendererInterblockSeparator
{}\markdownRendererCodeSpan{
source \markdownRendererTilde{}/.bashrc
plink --help
}\markdownRendererOlItemEnd 
\markdownRendererOlEnd \markdownRendererInterblockSeparator
{}
\markdownRendererSectionEnd \markdownRendererSectionBegin
\markdownRendererHeadingThree{Get TASSEL (GUI) on Linux}\markdownRendererInterblockSeparator
{}\markdownRendererOlBeginTight
\markdownRendererOlItemWithNumber{1}Go on the website \markdownRendererLink{https://www.maizegenetics.net/tassel}{https://www.maizegenetics.net/tassel}{https://www.maizegenetics.net/tassel}{} and download the last UNIX verison.\markdownRendererOlItemEnd 
\markdownRendererOlItemWithNumber{2}Download the TASSEL\markdownRendererUnderscore{}\markdownRendererLeftBrace{}xxx\markdownRendererRightBrace{}\markdownRendererUnderscore{}unix.sh and make it executable \markdownRendererCodeSpan{
   chmod +x \markdownRendererTilde{}/Downloads/TASSEL\markdownRendererUnderscore{}\markdownRendererLeftBrace{}xxx\markdownRendererRightBrace{}\markdownRendererUnderscore{}unix.sh
  }\markdownRendererOlItemEnd 
\markdownRendererOlItemWithNumber{3}Run the TASSEL installer \markdownRendererCodeSpan{
   \markdownRendererTilde{}/Downloads/TASSEL\markdownRendererUnderscore{}\markdownRendererLeftBrace{}xxx\markdownRendererRightBrace{}\markdownRendererUnderscore{}unix.sh
  }\markdownRendererOlItemEnd 
\markdownRendererOlEndTight \markdownRendererInterblockSeparator
{}
\markdownRendererSectionEnd \markdownRendererSectionBegin
\markdownRendererHeadingThree{Get \markdownRendererCodeSpan{rTASSEL} working in Linux}\markdownRendererInterblockSeparator
{}\markdownRendererOlBeginTight
\markdownRendererOlItemWithNumber{1}\markdownRendererCodeSpan{rJava} installation\markdownRendererOlItemEnd 
\markdownRendererOlEndTight \markdownRendererInterblockSeparator
{}\markdownRendererCodeSpan{
   sudo apt install default-jdk
   sudo R CMD javareconf
   R install.packages("rJava")
  }\markdownRendererInterblockSeparator
{}\markdownRendererOlBeginTight
\markdownRendererOlItemWithNumber{2}Installation in R\markdownRendererOlItemEnd 
\markdownRendererOlEndTight \markdownRendererInterblockSeparator
{}\markdownRendererCodeSpan{
   if (!require("devtools")) install.packages("devtools")
   devtools::install\markdownRendererUnderscore{}github(
    repo = "maize-genetics/rTASSEL",
    ref = "master",
    build\markdownRendererUnderscore{}vignettes = TRUE,
    dependencies = TRUE
   )
  }\markdownRendererInterblockSeparator
{}\markdownRendererOlBeginTight
\markdownRendererOlItemWithNumber{3}Run \markdownRendererCodeSpan{rTASSEL}\markdownRendererOlItemEnd 
\markdownRendererOlEndTight \markdownRendererInterblockSeparator
{}\markdownRendererUlBeginTight
\markdownRendererUlItem Allocate job's memory<sup>1</sup> and start the logger (here at the root of the project):\markdownRendererUlItemEnd 
\markdownRendererUlEndTight \markdownRendererInterblockSeparator
{}<sup>1</sup>"-Xmx50g" and "-Xms50g", "\markdownRendererEmphasis{50g}" represents 50 Gigabytes of memory.\markdownRendererInterblockSeparator
{}\markdownRendererEmphasis{!! Choose an appropriate value that fits your machine !!}\markdownRendererInterblockSeparator
{}\markdownRendererCodeSpan{
   options(java.parameters = c("-Xmx50g", "-Xms50g"))
   rTASSEL::startLogger(fullPath = NULL, fileName = NULL)
  }\markdownRendererInterblockSeparator
{}\markdownRendererUlBeginTight
\markdownRendererUlItem Run \markdownRendererAmpersand{} infos\markdownRendererUlItemEnd 
\markdownRendererUlEndTight \markdownRendererInterblockSeparator
{}\markdownRendererCodeSpan{
   library(rTASSEL)
   ??rTASSEL
  }\markdownRendererInterblockSeparator
{}Useful resource for \markdownRendererCodeSpan{rTASSEL} are the vignettes and tutorials at \markdownRendererLink{https://rtassel.maizegenetics.net/index.html}{https://rtassel.maizegenetics.net/index.html}{https://rtassel.maizegenetics.net/index.html}{}\markdownRendererInterblockSeparator
{}
\markdownRendererSectionEnd \markdownRendererSectionBegin
\markdownRendererHeadingThree{Get \markdownRendererCodeSpan{GEMMA}}\markdownRendererInterblockSeparator
{}GEMMA can be installed from source at the GitHub repo, but is also available through Bioconda \markdownRendererLink{http://www.ddocent.com/bioconda/}{http://www.ddocent.com/bioconda/}{http://www.ddocent.com/bioconda/}{}. To install is suggested to have miniconda installed and working, and then added the channel for Bioconda, you should already have defaults and conda-forge.\markdownRendererInterblockSeparator
{}\markdownRendererCodeSpan{
conda config --add channels defaults
conda config --add channels conda-forge
conda config --add channels biocond
conda install gemma
}\markdownRendererInterblockSeparator
{}And use GEMMA with\markdownRendererInterblockSeparator
{}\markdownRendererCodeSpan{
gemma -h
}\markdownRendererInterblockSeparator
{}\markdownRendererThematicBreak{}
\markdownRendererSectionEnd 
\markdownRendererSectionEnd 
\markdownRendererSectionEnd \markdownRendererDocumentEnd